File: Examenes/ExamenPSP/2DAM_PSP-main/9 - Caja fuerte asimétrica/CajaFuerte - SIN FIRMAR MENSAJES/cliente/src/main/java/ui/main/MainFX.java
package ui.main;

import jakarta.enterprise.event.Observes;
import jakarta.inject.Inject;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.stage.Stage;
import ui.screens.common.ScreenConstants;
import ui.screens.common.Screens;
import ui.screens.principal.PrincipalController;

import java.io.IOException;

public class MainFX {

    @Inject
    FXMLLoader fxmlLoader;

    public void start(@Observes @StartupScene Stage stage) {
        try {
            Parent fxmlParent = fxmlLoader.load(getClass().getResourceAsStream(Screens.PRINCIPAL.getRuta()));
            PrincipalController controller = fxmlLoader.getController();
            controller.setStage(stage);
            // TOMÁS - MODIFICAR MEDIDAS DE LA PANTALLA
            stage.setScene(new Scene(fxmlParent, 1050, 660));
            stage.setTitle(ScreenConstants.NEWSPAPERS_TOMAS);
            stage.show();
        } catch (IOException e) {
            e.printStackTrace();
            System.exit(0);
        }
    }

}

File: Examenes/ExamenPSP/2DAM_PSP-main/11 - Spring GraphQL/graphqlCliente/src/main/kotlin/MainFX.kt
import javafx.application.Application
import javafx.fxml.FXMLLoader
import javafx.scene.Scene
import javafx.scene.layout.BorderPane
import javafx.stage.Stage
import ui.common.Screens
import ui.controllers.PrincipalController


class MainFX : Application() {


    override fun start(primaryStage: Stage?) {
        try {
            val loaderMenu = FXMLLoader(
                javaClass.getResource(Screens.PRINCIPAL.ruta)
            )

            val root: BorderPane? = loaderMenu.load()
            val controller: PrincipalController = loaderMenu.getController()
            controller.setStage(primaryStage)
            val scene = Scene(root)
            primaryStage!!.title = "PersMascs!!!"
            primaryStage.scene = scene
            primaryStage.isResizable = true
            primaryStage.minWidth = 800.0
            primaryStage.minHeight = 1000.0
            primaryStage.isMaximized = true
            primaryStage.show()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

}

fun main(args: Array<String>) {
    Application.launch(MainFX::class.java, *args)
}
